options.go

```diff
@@ -0,0 +1,119 @@
+package asyncretry
+
+import (
+	"context"
+	"time"
+)
+
+// Option represents an option for async-retry.
+type Option func(*Config)
```

Comment:
[optional]

You chose FOP, I think it's a very good idea.
I have one suggestion for you.
How about using the Option interface that  holds an unexported methods and recording options of the unexported option struct, as in Uber's Go Style Guide?

https://github.com/uber-go/guide/blob/master/style.md#functional-options

The above pattern is more flexible for the author and easier for the user to debug and test.

Reply:
Thanks for information!
I did not know Uber's way. 
I read doc, it is very interesting, but I think  Uber's way is overkill for us.
I think pattern with closures is not bad for us, so I want to stay.

Reply:
OK!
I don't think I have enough of an advantage to fix it either.